### PR TITLE
MODULES-1622: Allow multiple Deny directives in a directory

### PR DESCRIPTION
**Pull request by faker-**
Allow multiple `Deny` directives in the same way as `Allow` already works, example:  

    directories => [
      { 'path' => '/var/ww/first',
        'deny' => [ 'from 127.0.0.1', 'from 127.0.0.2', ],
      },

### CODE REVIEW

**Comment by igalic** [comment]
@faker- i *think* we have tests for this.
would you mind updating those too?

**Comment by mhaskel** [comment]
@faker- there are some tests in https://github.com/puppetlabs/puppetlabs-apache/blob/master/spec/defines/vhost_spec.rb. It would be great if you could add a case with `deny` set to an array just to make sure nothing explodes. Otherwise this looks good. Thanks for the contribution!

**Comment by faker-** [comment]
Sorry for the delay, I had trouble setting up an environment for the tests. 
Can you check my last commit?
I'm not entirely my changes are all that is required.

**Comment by mhaskel** [comment]
@faker- if there could also be a test case with something like `deny => 'from all'` just to make sure no array-specific logic isn't properly wrapped that would be great. Thanks!

**Comment by faker-** [comment]
Makes sense, added with my last commit.

**Comment by igalic** [comment]
@faker- would you mind [squashing](http://www.git-scm.com/book/en/Git-Tools-Rewriting-History#Squashing-Commits) your commits down a bit?

**Comment by faker-** [comment]
@igalic squashed them to one commit

File: spec/defines/vhost_spec.rb [hunk 1]
```diff
@@ -147,11 +147,20 @@
...
+              'allow'    => [ 'from 127.0.0.1', 'from 127.0.0.2', ],
+              'deny'     => [ 'from 127.0.0.3', 'from 127.0.0.4', ],
+            },
+            {
+              'path'     => '/var/www/foo',
+              'provider' => 'files',
+              'allow'    => 'from 127.0.0.1',
+              'deny'     => 'from all',
+            },
+          ],
```
Comment (igalic) [inline spec/defines/vhost_spec.rb]: i just realized, you're not checking whether this sets or changes anything down the line

**Comment by faker-** [comment]
@igalic you are right, I just extended the existing "deny from all" test.
Turns out that deny was not getting applied anyway (since Apache version in that block is set to 2.4).

Can you check my last commit again?

**Comment by mhaskel** [comment]
Looks good, thanks @faker- !

**Pull request merged**